### PR TITLE
Merge 06-22

### PR DESCRIPTION
**Pull request by paulkaplan**
Merging from blockly develop to get variable event changes. 

A few things of note:

- I didn't take the change to gesture.js from https://github.com/google/blockly/issues/1109, since it breaks dropdowns in the flyout. Should I file an issue for that upstream?
- Since the flyout stuff was broken out in blockly, there were some (minor seeming) conflicts that I didn't know how to resolve. It seems that the flyout code may have drifted a bit. We may want to revisit that on the next merge / separately.
- We can finally take the `"Shadow blocks cannot have variable fields."` assertion! :)
- There were no merge conflicts in core/events, so i didn't look too closely at it, but you might want to pay special attention there to watch out for duplicates / bad merge issues.
- Ran my handy "duplicate function definition" script before submitting PR, so there should be less of that :)

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
Looks solid to me, though I'd like @marisaleung to double-check the events and variables stuff.  That's most of the merge this time.

Flyout code drifting--what in particular are you referring to?  Yes, we can look at that separately, but let's not wait until the next merge.

File: core/flyout_base.js [hunk 1]
```diff
@@ -278,6 +276,13 @@ Blockly.Flyout.prototype.init = function(targetWorkspace) {
   // A flyout connected to a workspace doesn't have its own current gesture.
   this.workspace_.getGesture =
       this.targetWorkspace_.getGesture.bind(this.targetWorkspace_);
+
+  // Get variables from the main workspace rather than the target workspace.
+  this.workspace_.getVariable =
+      this.targetWorkspace_.getVariable.bind(this.targetWorkspace_);
+
+  this.workspace_.getVariableById =
+      this.targetWorkspace_.getVariableById.bind(this.targetWorkspace_);
```
Comment (marisaleung) [inline core/flyout_base.js]: If it's possible, I'd add this PR https://github.com/google/blockly/pull/1181 to this merge. Sorry, I thought it had been merged into Blockly already.

File: core/workspace.js [hunk 2]
```diff
@@ -299,6 +296,7 @@ Blockly.Workspace.prototype.renameVariable = function(oldName, newName) {
   this.renameVariableInternal_(variable, newName);
 };
 
+
```
Comment (marisaleung) [inline core/workspace.js]: nit remove newline

**Comment by marisaleung** [comment]
Update on gesture breakage:
I filed a bug for it https://github.com/google/blockly/issues/1185
And fixed it https://github.com/google/blockly/pull/1186

**Comment by marisaleung** [comment]
Everything looks great! Also curious what you mean by flyout code drifting.

File: core/flyout_base.js (hunk 1 above)
Comment (paulkaplan) [reply core/flyout_base.js]: Fixed in a793911

File: core/workspace.js (hunk 2 above)
Comment (paulkaplan) [reply core/workspace.js]: Fixed with 	2af85c9

**Pull request merged**